scene/gui/button.cpp

```diff
@@ -439,7 +439,7 @@ Size2 Button::get_minimum_size_for_text_and_icon(const String &p_text, Ref<Textu
 		}
 	}
 
-	if (!xl_text.is_empty() || !p_text.is_empty()) {
+	if (minsize.height == 0.f && (!xl_text.is_empty() || !p_text.is_empty())) {
```

Comment:
Should this be an exact or a fuzzy check? In either case it should be `0.0` and not explicitly float, as `Size2` uses `real_t`

Reply:
Exact, since 0 means there was no text set.
Maybe this entire conditional should be deleted since when minsize.height is zero, xl_text and p_text are empty.  I'll try it out.